### PR TITLE
"[LF] drop transaction version < 14

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

**Review by paulbrauner-da — APPROVED** [review]
Nice cleanup! Thank you for also updating the spec.

File: daml-lf/spec/transaction.rst [hunk 1]
```diff
@@ -461,54 +436,25 @@ fields are required, and required to be non-empty.
 
 ``children`` is constrained as described under `field node_id`_.
 
+Every element of ``actors``, ``stakeholders``, ``signatories``, and
+``controllers``, and ``observers`` must be a party identifier.
+
+``arg_unversioned`` and ``result_unversioned`` are required, while
+``arg_versioned`` and ``result_versioned`` must be used.
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/spec/transaction.rst:443]: Should be "unused".
Maybe I should use something else, e.g. "undefined", "unset" ?

[the remark this answers is not in the document]
Comment (paulbrauner-da) [reply daml-lf/spec/transaction.rst:443]: Yeah, "unset" or "should not be set" sound good to me.

File: daml-lf/spec/transaction.rst [hunk 2]
```diff
@@ -228,9 +221,9 @@ message ContractInstance
 An instance of a Daml-LF template, represented by the Daml-LF value used
 to construct that instance.
 
-(*since version 10*)
+(*since version 14*)
 
-As of version 10, these fields are included:
+As of version 14, these fields are included:
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/spec/transaction.rst:226]: This spec does not speak anymore of version 10.

[the remark this answers is not in the document]
Comment (paulbrauner-da) [reply daml-lf/spec/transaction.rst:226]: Then I think you want to fix line 249.

**Pull request merged**